Exception during initialisation may leave the thing ONLINE

```
2024-10-07 10:21:24.896 [ERROR] [ng.zigbee.handler.ZigBeeThingHandler] - A4C138864DFA2AE2: Exception creating channels 
java.lang.ClassCastException: null
==> /var/log/openhab/events.log <==
2024-10-07 10:21:24.897 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'zigbee:device:house:SmartPlug_ElectricBlanket' changed from UNKNOWN to OFFLINE (HANDLER_INITIALIZING_ERROR)
2024-10-07 10:21:24.898 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'zigbee:device:house:SmartPlug_ElectricBlanket' changed from OFFLINE (HANDLER_INITIALIZING_ERROR) to ONLINE
```

--- a/org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java
+++ b/org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java
@@ -473,13 +473,12 @@ public class ZigBeeThingHandler extends BaseThingHandler implements ZigBeeNetwor
             otaServer.addListener(this);
         }
 
-        updateStatus(ThingStatus.ONLINE);
-
         startPolling();
 
         nodeInitialised = true;
 
         logger.debug("{}: Done initialising ZigBee Thing handler", nodeIeeeAddress);
+        updateStatus(ThingStatus.ONLINE);
 
         // Save the network state
         coordinatorHandler.serializeNetwork(node.getIeeeAddress());
@@ -518,7 +517,9 @@ public class ZigBeeThingHandler extends BaseThingHandler implements ZigBeeNetwor
      * If the channel converter responds to a command, then the thing is considered ONLINE
      */
     public void alive() {
-        updateStatus(ThingStatus.ONLINE);
+        if (nodeInitialised) {
+            updateStatus(ThingStatus.ONLINE);
+        }
     }
 
     private synchronized void initializeDevice() {
@@ -811,7 +812,9 @@ public class ZigBeeThingHandler extends BaseThingHandler implements ZigBeeNetwor
         }
         logger.debug("{}: Updating ZigBee channel state {} to {}", nodeIeeeAddress, channel, state);
         updateState(channel, state);
-        updateStatus(ThingStatus.ONLINE);
+        if (nodeInitialised) {
+            updateStatus(ThingStatus.ONLINE);
+        }
         isAliveTracker.resetTimer(this);
     }
 
@@ -830,8 +833,10 @@ public class ZigBeeThingHandler extends BaseThingHandler implements ZigBeeNetwor
             return;
         }
         logger.debug("{}: Triggering ZigBee channel {} with event {}", nodeIeeeAddress, channel, event);
-        super.triggerChannel(channel, event);
-        updateStatus(ThingStatus.ONLINE);
+        triggerChannel(channel, event);
+        if (nodeInitialised) {
+            updateStatus(ThingStatus.ONLINE);
+        }
         isAliveTracker.resetTimer(this);
     }
 

--- a/org.openhab.binding.zigbee/src/test/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandlerTest.java
+++ b/org.openhab.binding.zigbee/src/test/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandlerTest.java
@@ -29,6 +29,7 @@ import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 
 import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
 import org.openhab.binding.zigbee.ZigBeeBindingConstants;
 import org.openhab.binding.zigbee.converter.ZigBeeBaseChannelConverter;
@@ -37,7 +38,10 @@ import org.openhab.core.config.core.Configuration;
 import org.openhab.core.thing.Channel;
 import org.openhab.core.thing.ChannelUID;
 import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusInfo;
 import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandlerCallback;
 import org.openhab.core.thing.type.ThingType;
 import org.openhab.core.thing.type.ThingTypeBuilder;
 
@@ -65,6 +69,18 @@ public class ZigBeeThingHandlerTest {
         privateMethod.setAccessible(true);
 
         return (List<Integer>) privateMethod.invoke(handler, initialClusters, newClusters);
+    }
+
+    @Test
+    public void testAliveWhenNotInitialized() {
+        ZigBeeThingHandler handler = new ZigBeeThingHandler(null, null, null);
+
+        ThingHandlerCallback callback = mock(ThingHandlerCallback.class);
+        handler.setCallback(callback);
+
+        handler.alive();
+
+        verify(callback, never()).statusUpdated(any(), any());
     }
 
     @Test
@@ -160,6 +176,9 @@ public class ZigBeeThingHandlerTest {
         injectIntoPrivateField(zigBeeThingHandler, zigBeeCoordinatorHandler, "coordinatorHandler");
         injectIntoPrivateField(zigBeeThingHandler, ieeeAddress, "nodeIeeeAddress");
 
+        ThingHandlerCallback callback = mock(ThingHandlerCallback.class);
+        zigBeeThingHandler.setCallback(callback);
+
         // call doNodeInitialisation by reflection as it is not accessible
         Method doNodeInitialisationMethod = ZigBeeThingHandler.class.getDeclaredMethod("doNodeInitialisation",
                 (Class<Object>[]) null);
@@ -177,6 +196,13 @@ public class ZigBeeThingHandlerTest {
             Mockito.verify(thing, times(1)).setProperty(ZigBeeBindingConstants.THING_PROPERTY_DEVICE_INITIALIZED,
                     Boolean.TRUE.toString());
         }
+
+        ArgumentCaptor<ThingStatusInfo> captor = ArgumentCaptor.forClass(ThingStatusInfo.class);
+        verify(callback, times(1)).statusUpdated(any(), captor.capture());
+
+        ThingStatusInfo status = captor.getValue();
+        assertEquals(ThingStatus.ONLINE, status.getStatus());
+
     }
 
     private ZigBeeChannelConverterFactory mockZigBeeChannelConverterFactory(

